gtk3/lib/gtk3/tree-iter.rb

```diff
@@ -73,6 +73,10 @@ def children
       @model.iter_children(self)
     end
 
+    def first_child
+      @model.iter_nth_child(self, 0)
```

Comment:
Yes. Your implementation is more efficient. There is no magic.

> Or maybe you think that the readability if more important that the efficiency we could add?

Yes. I choose maintainability and readability in this case.

With your implementation, if we need to change `#nth_child` implementation, we also need to change `#first_child` implementation. (maintainability)

The current implementation says "the first child is 0th child". It's straightforward. (readability)

> As I have said before I will do what you asked me to do, I just try to have a better understanding of what I am doing.

I like your style. :-)
If you have any question in the future, please ask me as usual. :-)